TransferMethod/Sources/UpdateTransferMethodPresenter.swift

```diff
@@ -0,0 +1,313 @@
+    private weak var view: UpdateTransferMethodView?
+    private let errorTypeApi = "API"
+    private let createdConfirmationPageName = "transfer-method:update:transfer-method-updated"
+    private let pageLink = "update-transfer-method"
+    private let transferMethodUpdatedGoal = "transfer-method-updated"
+    static let updateTransferMethodPageGroup = "update-transfer-method"
+    static let updateTransferMethodPageName = "transfer-method:update:collect-transfer-method-information"
+    private let hyperwalletInsights: HyperwalletInsightsProtocol
+    let transferMethod: HyperwalletTransferMethod?
+    var sectionData = [UpdateTransferMethodSectionData]()
+
+    private lazy var transferMethodUpdateConfigurationRepository = {
+        TransferMethodRepositoryFactory.shared.transferMethodUpdateConfigurationRepository()
+    }()
+
+    private lazy var transferMethodRepository = {
+        TransferMethodRepositoryFactory.shared.transferMethodRepository()
+    }()
+
+    init(_ view: UpdateTransferMethodView,
+         _ transferMethod: HyperwalletTransferMethod,
+         _ hyperwalletInsights: HyperwalletInsightsProtocol = HyperwalletInsights.shared) {
+        self.view = view
+        self.transferMethod = transferMethod
+        self.hyperwalletInsights = hyperwalletInsights
+    }
+
+    func loadTransferMethodUpdateConfigurationFields(_ forceUpdate: Bool = false) {
+        view?.showLoading()
+
+        if forceUpdate {
+            transferMethodUpdateConfigurationRepository.refreshFields()
+        }
+
+        transferMethodUpdateConfigurationRepository
+            .getFields(transferMethod?.token ?? "") { [weak self] (result) in
+                guard let strongSelf = self, let view = strongSelf.view else {
+                    return
+                }
+                view.hideLoading()
+
+                switch result {
+                case .failure(let error):
+                    view.showError(
+                        error,
+                        pageName: UpdateTransferMethodPresenter.updateTransferMethodPageName,
+                        pageGroup: UpdateTransferMethodPresenter.updateTransferMethodPageGroup) {
+                            strongSelf.loadTransferMethodUpdateConfigurationFields()
+                    }
+
+                case .success(let fieldResult):
+                    if let fieldGroups = fieldResult?.transferMethodUpdateConfiguration()?
+                        .fieldGroups?.nodes {
+                        strongSelf.trackUILoadImpression()
+                        view.reloadData(fieldGroups)
+                    }
+                }
+            }
+    }
+
+    func updateTransferMethod() {
+        trackConfirmClick()
+        guard let view = view, view.areAllFieldsValid() else {
+            return
+        }
+
+        guard let transferMethod = transferMethod,
+            let hyperwalletTransferMethod = buildHyperwalletTransferMethod(transferMethod) else {
```

Comment:
since you have just transferMethodToken and hyperwalletTransferMethodConfiguration, use these both to buildHyperwalletTransferMethod

Reply:
Updated.